src/main/java/org/opentripplanner/routing/graph/Graph.java

```diff
@@ -215,20 +216,38 @@
     /** Parent stops **/
     public Map<FeedScopedId, Stop> parentStopById = new HashMap<>();
 
-    /** Data model for Raptor routing */
+    // TripPatterns used to be reached through hop edges, but we're not creating on-board transit vertices/edges anymore.
+    public Map<String, TripPattern> tripPatternForId = Maps.newHashMap();
+
+    /** Interlining relationships between trips. */
+    public final BiMap<Trip,Trip> interlinedTrips = HashBiMap.create();
+
+    /** Data model for Raptor routing, with realtime updates applied (if any). */
     public transient TransitLayer transitLayer;
 
+    /** Data model for Raptor routing, with realtime updates applied (if any). */
+    public transient TransitLayer realtimeTransitLayer;
+
+    // Hack. I've tried three different ways of generating unique labels.
+    // Previously we were just tolerating edge label collisions.
+    // For some reason we're repeatedly generating splits on the same edge objects, despite a comment that said it was
+    // guaranteed there would only ever be one split per edge. This is going to fail as soon as we load a base OSM graph
+    // and build transit on top of it.
+    public long nextSplitNumber = 0;
+
```

Comment:
The Graph definitely needs to be restructured and broken down. However rather than trying to rework the existing graph, I'd be more in favor of creating the better layout when creating the new transit data classes we talked about earlier (similar richness to existing OTP1 data, but implementing OTP2 raptor interfaces).

Reply:
Fixed.